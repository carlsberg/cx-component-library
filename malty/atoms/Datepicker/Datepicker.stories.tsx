import { Meta, Story } from '@storybook/react';
import React, { useState } from 'react';
import { Datepicker as DatepickerComponent } from './Datepicker';
import { DatepickerProps } from './Datepicker.types';

export default {
  title: 'Forms/Datepicker',
  component: DatepickerComponent,
  parameters: {
    importObject: 'Datepicker',
    importPath: '@carlsberggroup/malty.atoms.datepicker'
  },
  argTypes: {
    label: {
      description: 'The datepicker input label',
      control: {
        type: 'string'
      }
    },
    startDate: {
      description: 'Initial selected date',
      control: {
        type: 'date'
      }
    },
    onChange: {
      description: 'Action to perform when clicking a calendar day'
    },
    minDate: {
      description: 'disable days before defined min. date',
      control: {
        type: 'date'
      }
    },
    maxDate: {
      description: 'disable days after defined max. date',
      control: {
        type: 'date'
      }
    },
    excludeDates: {
      description: 'disable array of days',
      control: {
        type: 'array'
      }
    },
    disabled: {
      description: 'disable datepicker',
      control: {
        type: 'boolean'
      }
    },
    locale: {
      description: 'iso language code',
      control: {
        type: 'string'
      }
    }
  }
} as Meta;

const Template: Story<DatepickerProps> = ({ label }) => {
  const [startDate, setStartDate] = useState(new Date());

  return (
    <div style={{ height: '400px' }}>
      <DatepickerComponent label={label} onChange={(date: Date) => setStartDate(date)} startDate={startDate} />
    </div>
  );
};

export const Datepicker = Template.bind({});
Datepicker.args = { label: 'Select date' };
