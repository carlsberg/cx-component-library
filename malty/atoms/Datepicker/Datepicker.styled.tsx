import styled from 'styled-components';

export const StyledDatepicker = styled.div`
  position: relative;
  & .react-datepicker-popper {
    width: 100%;
    max-width: 544px;
    min-width: 304px;
  }
`;

export const StyledInput = styled.div<{
  disabled?: boolean;
}>`

  display: flex;
  flex-direction: column;
  width: 100%;
  & label {
    font-size: ${({ theme }) => theme.typography.desktop.text.small_bold['font-size'].value};
    font-weight: ${({ theme }) => theme.typography.desktop.text.small_bold['font-weight'].value};
    display: block;
    margin-bottom: ${({ theme }) => theme.sizes['2xs'].value};
    &:first-letter {
        text-transform: capitalize;
    }
  }
  & button {
    position: relative;
    padding: 0;
    font: inherit;
    color: inherit;
    background-color: transparent;
    cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
    height: ${({ theme }) => theme.sizes['2xl'].value};
    width: 100%;
    min-width: 150px;
    box-sizing: border-box;
    text-align: left;
    padding: 0 ${({ theme }) => theme.sizes.s.value};
    border-width: ${({ theme }) => `${theme.borders['border-1px--solid']['border-width'].value}`};
    border-style:  ${({ theme }) => `${theme.borders['border-1px--solid']['border-style'].value}`};
    border-color: ${({ disabled, theme }) =>
      `${disabled ? theme.colors.colours.support['40'].value : theme.colors.colours.default['digital-black'].value}`};
    font-size: ${({ theme }) => theme.typography.desktop.text['medium-small_bold']['font-size'].value};
  }
  & svg {
      position: absolute;
      top: 50%;
      transform: translateY(-50%);
      right: ${({ theme }) => theme.sizes.s.value};
  }
}
`;

export const StyledContainer = styled.div`
  //place container on top of datepicker input border
  margin-top: -1px;
  box-sizing: border-box;
  border: ${({ theme }) => `${theme.borders['border-1px--solid']['border-width'].value}
  ${theme.borders['border-1px--solid']['border-style'].value}
  ${theme.colors.colours.default['digital-black'].value}`};
  width: 100%;
  user-select: none;
  padding: ${({ theme }) => `${theme.sizes.xl.value} ${theme.sizes.m.value} ${theme.sizes.m.value}`};
`;

export const StyledCalendar = styled.div`
  display: flex;
  font-size: ${({ theme }) => theme.typography.desktop.text.small_default['font-size'].value};
  position: relative;
  & .react-datepicker__header {
    background-color: transparent;
    border-bottom: none;
    text-align: left;
  }
  & .react-datepicker__current-month {
    font-size: ${({ theme }) => theme.typography.desktop.text.medium_default['font-size'].value};
    font-weight: ${({ theme }) => theme.typography.desktop.headline.medium['font-weight'].value};
  }
  & .react-datepicker__day-names {
    font-weight: ${({ theme }) => theme.typography.desktop.text.small_bold['font-weight'].value};
    width: 100%;
    margin-top: ${({ theme }) => theme.sizes.l.value};
    margin-bottom: ${({ theme }) => theme.sizes.m.value};
    background-color: ${({ theme }) => theme.colors.colours.support['20'].value};
    justify-content: space-between;
    display: flex;
    padding: 0  ${({ theme }) => theme.sizes.xs.value};
    box-sizing: border-box;
  }
  & .react-datepicker__day-name {
    color: ${({ theme }) => theme.colors.colours.support['60'].value};
    width: auto;
    display: inline-block;
    width: ${({ theme }) => theme.sizes.l.value};
    line-height: ${({ theme }) => theme.sizes.l.value};
    text-align: center;
    margin: 0.166rem;
    
  }
  & .react-datepicker__month-container {
      width 100%;
  }
  & .react-datepicker__month {
    padding: 0  ${({ theme }) => theme.sizes.xs.value};
  }
  & .react-datepicker__week {
      display: flex;
      justify-content: space-between;
      margin-bottom: ${({ theme }) => theme.sizes['4xs'].value};
      white-space: nowrap;
  }
  & .react-datepicker__day {
      text-align: center;
      cursor: pointer;
      width: ${({ theme }) => theme.sizes.l.value};
      height: ${({ theme }) => theme.sizes.l.value};
      border-radius: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
      box-sizing: border-box;
      font-weight: ${({ theme }) => theme.typography.desktop.text.small_bold['font-weight'].value};
      &:hover {
        background-color: ${({ theme }) => theme.colors.colours.support['20'].value};
      }
  }
  & .react-datepicker__day--today {
      border: ${({ theme }) => `${theme.borders['border-1px--solid']['border-width'].value}
      ${theme.borders['border-1px--solid']['border-style'].value}
      ${theme.colors.colours.default['digital-black'].value}`};
      &:hover {
        background-color: inherit;
      }
  }
  & .react-datepicker__day--selected {
    background-color: ${({ theme }) => theme.colors.colours.default['digital-black'].value};
    color: ${({ theme }) => theme.colors.colours.default.white.value};
    &:hover {
      background-color: ${({ theme }) => theme.colors.colours.default['digital-black'].value};
    }
}
  & .react-datepicker__navigation {
    width: ${({ theme }) => theme.sizes.l.value};
    height: ${({ theme }) => theme.sizes.l.value};
    align-items: center;
    background: none;
    display: flex;
    justify-content: center;
    text-align: center;
    cursor: pointer;
    position: absolute;
    padding: 0;
    border: none;
    z-index: 1;
    text-indent: -999em;
    overflow: hidden;
    top: ${({ theme }) => `-${theme.sizes['2xs'].value}`};
  }
  & .react-datepicker__navigation-icon {
    position: relative;
    top: -1px;
    font-size: ${({ theme }) => theme.typography.desktop.headline.large['font-size'].value};
    width: 0;
    &:before {
        border-radius: 1px;
        border-width: ${({ theme }) =>
          `${theme.borders['border-2px--solid']['border-width'].value} ${theme.borders['border-2px--solid']['border-width'].value} 0 0`};
        border-color: ${({ theme }) => theme.colors.colours.default['digital-black'].value};
        content: "";
        display: block;
        height: ${({ theme }) => theme.sizes['2xs'].value};
        position: absolute;
        top: ${({ theme }) => theme.sizes['3xs'].value};
        width: ${({ theme }) => theme.sizes['2xs'].value};
        border-style: solid;
    }
  }
  & .react-datepicker__day--disabled {
    font-weight: ${({ theme }) => theme.typography.desktop.text.small_default['font-weight'].value};
    color: ${({ theme }) => theme.colors.colours.support['60'].value};
  }
  & .react-datepicker__navigation--previous {
      left: initial;
      right: ${({ theme }) => theme.sizes['4xl'].value};
  }
  & .react-datepicker__navigation-icon--previous {
    &:before {
      transform: rotate(225deg);
    }
  }
  & .react-datepicker__navigation--next {
    left: initial;
    right: ${({ theme }) => theme.sizes.xs.value};
  }
  & .react-datepicker__navigation-icon--next {
    &:before {
      transform: rotate(45deg);
      left: ${({ theme }) => `-${theme.sizes.xs.value}`};
    }
  }
`;
