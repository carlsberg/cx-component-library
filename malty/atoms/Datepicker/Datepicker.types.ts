export interface DatepickerProps {
  label: string;
  startDate: Date;
  onChange: (date: Date) => void;
}
