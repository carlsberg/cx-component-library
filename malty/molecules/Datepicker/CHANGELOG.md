# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.2.0 (2024-05-31)

### Bug Fixes

- **datepicker:** [DSM-547] Datepicker unwanted text ([#560](https://github.com/CarlsbergGBS/cx-component-library/issues/560)) ([3e3e31e](https://github.com/CarlsbergGBS/cx-component-library/commit/3e3e31e31b45c51f6e6156dfcadf1bac1668a095))
- **datepicker:** [DSM-572] Add readonly on input when selectsRange ([#617](https://github.com/CarlsbergGBS/cx-component-library/issues/617)) ([58e47fc](https://github.com/CarlsbergGBS/cx-component-library/commit/58e47fc3c33d97e92d30d119b4a65063d019aa6b))
- **datepicker:** [DSM-824] add dateFormat prop to the component ([#833](https://github.com/CarlsbergGBS/cx-component-library/issues/833)) ([1240318](https://github.com/CarlsbergGBS/cx-component-library/commit/12403188db910f018c2c669f5911b007dcc808dd))
- **DSM-235:** fixed cursor when readOnly ([#201](https://github.com/CarlsbergGBS/cx-component-library/issues/201)) ([b3798ec](https://github.com/CarlsbergGBS/cx-component-library/commit/b3798ecb04dd1806f85fd877f271661457f26825))
- **DSM-312:** fixed rounded corners on mobile ([#250](https://github.com/CarlsbergGBS/cx-component-library/issues/250)) ([8eff302](https://github.com/CarlsbergGBS/cx-component-library/commit/8eff302cde8ad59e993cf082ddc8f68e383e9e54))
- **DSM-315:** fix rounded corners ([#272](https://github.com/CarlsbergGBS/cx-component-library/issues/272)) ([396a173](https://github.com/CarlsbergGBS/cx-component-library/commit/396a1732011c8936140ecee20070612230ec2444))
- **DSM-387:** Added missing onClick handler in buttons and opening and closing logic ([#315](https://github.com/CarlsbergGBS/cx-component-library/issues/315)) ([e9e8aed](https://github.com/CarlsbergGBS/cx-component-library/commit/e9e8aed6f0320b90bd58ba67aebd42d0fef04456))
- **DSM-411:** fixed datepicker minor bugs ([#340](https://github.com/CarlsbergGBS/cx-component-library/issues/340)) ([5303712](https://github.com/CarlsbergGBS/cx-component-library/commit/53037129cc406b8ae90f557b28b6af8c75739975))
- Minor component fixes ([#265](https://github.com/CarlsbergGBS/cx-component-library/issues/265)) ([c75bd7c](https://github.com/CarlsbergGBS/cx-component-library/commit/c75bd7c926163db6efb65ab89a64d8637d7a4664))
- **pagination:** [DSM-589] Refactor pagination and add validations ([#629](https://github.com/CarlsbergGBS/cx-component-library/issues/629)) ([caa545d](https://github.com/CarlsbergGBS/cx-component-library/commit/caa545dad76254fcc0893dc4c76cf0792ae40751))
- Remove typography provider from every component ([#383](https://github.com/CarlsbergGBS/cx-component-library/issues/383)) ([5d05680](https://github.com/CarlsbergGBS/cx-component-library/commit/5d05680e40d3da5bb3042bd8191427a70606afc3))
- reverted changes ([#396](https://github.com/CarlsbergGBS/cx-component-library/issues/396)) ([ecfcf57](https://github.com/CarlsbergGBS/cx-component-library/commit/ecfcf573863253fea16e24764b6f452c3cc4f215))
- Run auto fixable lint based on new `eslint` configuration ([#501](https://github.com/CarlsbergGBS/cx-component-library/issues/501)) ([9ee94cc](https://github.com/CarlsbergGBS/cx-component-library/commit/9ee94cc2cd407f717e62d8857b5f20a74e7bbec4))

### Features

- add tsconfig files and update package.json files for each component ([3bfc644](https://github.com/CarlsbergGBS/cx-component-library/commit/3bfc644e1cfc9dbb7cf7a0469e25fce055b53240))
- additional stories ([#291](https://github.com/CarlsbergGBS/cx-component-library/issues/291)) ([438af3b](https://github.com/CarlsbergGBS/cx-component-library/commit/438af3b969e0f5b4b908436ee58b93c4b9ddc97c))
- change organization name ([a0aa428](https://github.com/CarlsbergGBS/cx-component-library/commit/a0aa428f43138af5707a3ff4c0e36b7c056f02e6))
- **datepicker:** [DSM-799] add tests ([#783](https://github.com/CarlsbergGBS/cx-component-library/issues/783)) ([84ab41d](https://github.com/CarlsbergGBS/cx-component-library/commit/84ab41d714bf280d29454c33c9f50e6bba29ed63))
- **datepicker:** [DSM-967] add support for `onClose` callback ([#853](https://github.com/CarlsbergGBS/cx-component-library/issues/853)) ([6226f88](https://github.com/CarlsbergGBS/cx-component-library/commit/6226f8873cc77e92bef258b12c86562ad06de3ae))
- **DSM-119:** Datepicker background bug fix ([#164](https://github.com/CarlsbergGBS/cx-component-library/issues/164)) ([bc14919](https://github.com/CarlsbergGBS/cx-component-library/commit/bc14919e487ec9051d78e535adb0a0653385c0fc))
- **DSM-228:** Added rating component ([#379](https://github.com/CarlsbergGBS/cx-component-library/issues/379)) ([ab47354](https://github.com/CarlsbergGBS/cx-component-library/commit/ab47354db803fb051a1f87d66ee34966698df8bd))
- **DSM-235:** added readOnly on inputs ([#181](https://github.com/CarlsbergGBS/cx-component-library/issues/181)) ([36c5ef7](https://github.com/CarlsbergGBS/cx-component-library/commit/36c5ef7cb1225bf5a150918e01e10fd253377110))
- **DSM-248:** Headline HTML tags ([#184](https://github.com/CarlsbergGBS/cx-component-library/issues/184)) ([305f5e0](https://github.com/CarlsbergGBS/cx-component-library/commit/305f5e0fa92d5813f4877b338ad827a8c347d452))
- **DSM-362:** snapshots removed & unit test updates ([#307](https://github.com/CarlsbergGBS/cx-component-library/issues/307)) ([78423eb](https://github.com/CarlsbergGBS/cx-component-library/commit/78423eb5cfe04fe8a3a7af1f5af074ac6d35e235))
- **DSM-368:** Required prop & label component ([#384](https://github.com/CarlsbergGBS/cx-component-library/issues/384)) ([983bd78](https://github.com/CarlsbergGBS/cx-component-library/commit/983bd78c8f526a8faa2df601bf918b40219820f2))
- **DSM-385:** added dropdown to select year ([#368](https://github.com/CarlsbergGBS/cx-component-library/issues/368)) ([fe53588](https://github.com/CarlsbergGBS/cx-component-library/commit/fe53588f2142217b516aad4b71b0f934b391208e))
- **DSM-387:** datepicker include actions and captions ([#308](https://github.com/CarlsbergGBS/cx-component-library/issues/308)) ([eafd4ba](https://github.com/CarlsbergGBS/cx-component-library/commit/eafd4ba9a4b1a28d73dc0c569051548445aaad1a))
- **DSM-439:** Change font-family behavior ([#470](https://github.com/CarlsbergGBS/cx-component-library/issues/470)) ([ab6dad4](https://github.com/CarlsbergGBS/cx-component-library/commit/ab6dad414c42c4343de3de6abaa4cd3ff98cfca1))
- **DSM-513:** Add new behaviors in Datepicker ([#512](https://github.com/CarlsbergGBS/cx-component-library/issues/512)) ([098378c](https://github.com/CarlsbergGBS/cx-component-library/commit/098378c23b8e5d42359d26a3896b7a7f888e8d43))
- **general:** [DSM-953] refactor icons implementation ([#892](https://github.com/CarlsbergGBS/cx-component-library/issues/892)) ([08d259f](https://github.com/CarlsbergGBS/cx-component-library/commit/08d259f476ce97e9db8ac41d5eb4773eabca0d37))
- **global:** [DSM-784] add space props in all components ([#842](https://github.com/CarlsbergGBS/cx-component-library/issues/842)) ([9f03f59](https://github.com/CarlsbergGBS/cx-component-library/commit/9f03f59e7518a99abcea06bacbee989d1ddb0465))
- **global:** migrate from bit to lerna ([a0110c8](https://github.com/CarlsbergGBS/cx-component-library/commit/a0110c8831370dc762c193b17cc593eed381f990))
- **global:** Remove `react/jsx-props-no-spreading` comments ([#558](https://github.com/CarlsbergGBS/cx-component-library/issues/558)) ([823444c](https://github.com/CarlsbergGBS/cx-component-library/commit/823444ce0800bc7f49b584de59d247eb8497f559))
- **global:** update storybook files to be compliant with v7 ([#970](https://github.com/CarlsbergGBS/cx-component-library/issues/970)) ([c61d4b4](https://github.com/CarlsbergGBS/cx-component-library/commit/c61d4b40a0755becf942ad3f28758a159f8c54e4))
- new icon colors ([#215](https://github.com/CarlsbergGBS/cx-component-library/issues/215)) ([9247d33](https://github.com/CarlsbergGBS/cx-component-library/commit/9247d3370bcde78432033ba8b23837f97e0b0dc0))
- remove empty composition files ([f8173f4](https://github.com/CarlsbergGBS/cx-component-library/commit/f8173f4a2ecbf80bb7b6ffe848c023ae31819c2d))
- Several component updates ([#185](https://github.com/CarlsbergGBS/cx-component-library/issues/185)) ([633c9f0](https://github.com/CarlsbergGBS/cx-component-library/commit/633c9f0a4432e9e3d08ff05156453660929c19a2))
- Use new `eslint-config-carlsberg` with streamlined way to create components ([#559](https://github.com/CarlsbergGBS/cx-component-library/issues/559)) ([a459332](https://github.com/CarlsbergGBS/cx-component-library/commit/a45933215bc0b523220743f17d988d7f1ad5fce6))
- v1.0.22 ([#194](https://github.com/CarlsbergGBS/cx-component-library/issues/194)) ([885949f](https://github.com/CarlsbergGBS/cx-component-library/commit/885949fdf10b8883c0f4f45447fc5e5711bc5b3f))
