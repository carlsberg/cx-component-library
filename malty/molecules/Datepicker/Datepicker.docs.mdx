---
labels: ['react', 'typescript', 'datepicker']
description: 'A Datepicker component.'
---
