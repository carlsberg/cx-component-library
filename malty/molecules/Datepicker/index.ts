export { Datepicker } from './Datepicker';
export { Colors } from './Datepicker.types';
export type { DatepickerProps } from './Datepicker.types';
