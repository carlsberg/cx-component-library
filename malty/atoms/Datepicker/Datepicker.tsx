import { IconColor, IconSize } from '@carlsberggroup/malty.atoms.icon-wrapper';
import Calendar from '@carlsberggroup/malty.icons.calendar';
import { globalTheme as defaultTheme, TypographyProvider } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { ForwardedRef, forwardRef, useContext } from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import { ThemeContext } from 'styled-components';
import { StyledDatepicker, StyledInput } from './Datepicker.styled';
import { DatepickerProps } from './Datepicker.types';

export const Datepicker = ({ startDate, onChange, label }: DatepickerProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;

  const CustomInput = forwardRef(
    ({ value, onClick }: { value: Date; onClick: () => void }, ref: ForwardedRef<HTMLButtonElement>) => (
      <StyledInput theme={theme}>
        <label htmlFor="datepicker-input" onClick={onClick}>
          teste label {label}
        </label>
        <button name="datepicker-input" type="button" className="datepickerInput" onClick={onClick} ref={ref}>
          {value}
          <Calendar size={IconSize.Medium} color={IconColor.Primary} />
        </button>
      </StyledInput>
    )
  );

  return (
    <TypographyProvider>
      <StyledDatepicker>
        <DatePicker selected={startDate} onChange={(date: Date) => onChange(date)} customInput={<CustomInput />} />
      </StyledDatepicker>
    </TypographyProvider>
  );
};
