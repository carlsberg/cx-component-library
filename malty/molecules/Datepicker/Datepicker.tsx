import { IconColor, IconSize } from '@carlsberggroup/malty.atoms.icon-wrapper';
import Calendar from '@carlsberggroup/malty.icons.calendar';
import { globalTheme as defaultTheme, TypographyProvider } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { ReactNode, useContext } from 'react';
import DatePicker from 'react-datepicker';
import { ThemeContext } from 'styled-components';
import { Button, ButtonStyle } from '../../atoms/Button';
import { Text, TextStyle } from '../../atoms/Text';
import {
  StyledActionsContainer,
  StyledCalendar,
  StyledCaption,
  StyledCaptionContainer,
  StyledContainer,
  StyledDatepicker,
  StyledInputIcon,
  StyledLabel,
  StyledWrapper
} from './Datepicker.styled';
import { Caption, DatepickerProps } from './Datepicker.types';

export const Datepicker = ({
  startDate,
  endDate,
  onChange,
  label,
  locale,
  minDate,
  maxDate,
  disabled,
  excludeDates,
  selectsRange,
  inline,
  readOnly,
  ...props
}: DatepickerProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;

  const Container = ({ children }: { children: ReactNode }) => (
    <StyledContainer theme={theme}>
      <StyledCalendar theme={theme}>{children}</StyledCalendar>
    </StyledContainer>
  );

  // eslint-disable-next-line @typescript-eslint/no-shadow
  const renderDatepickerCaptions = (captions: Caption[]) => (
    <StyledCaptionContainer>
      {captions.map((caption, index) => (
        // eslint-disable-next-line react/no-array-index-key
        <StyledCaption color={caption.color} border={caption.border} key={`datepicker-caption-${index}`}>
          <Text textStyle={TextStyle.SmallDefault}>{caption.copy}</Text>
        </StyledCaption>
      ))}
    </StyledCaptionContainer>
  );

  const renderActions = () => (
    <StyledActionsContainer>
      {props.secondaryAction && <Button style={ButtonStyle.Secondary} text={props.secondaryAction.copy} fullWidth />}
      {props.primaryAction && <Button style={ButtonStyle.Primary} text={props.primaryAction.copy} fullWidth />}
    </StyledActionsContainer>
  );

  return (
    <TypographyProvider>
      <StyledWrapper theme={theme}>
        {!inline && (
          <StyledLabel disabled={disabled} htmlFor="datepicker-input" theme={theme}>
            {label}
          </StyledLabel>
        )}
        <StyledDatepicker disabled={disabled} readOnly={readOnly} theme={theme}>
          {!inline && (
            <StyledInputIcon disabled={disabled} readOnly={readOnly} theme={theme}>
              <Calendar size={IconSize.Medium} color={IconColor.DigitalBlack} />
            </StyledInputIcon>
          )}

          <DatePicker
            selected={startDate}
            startDate={startDate}
            endDate={selectsRange ? endDate : null}
            disabled={disabled}
            readOnly={readOnly}
            onChange={onChange}
            locale={locale}
            showPopperArrow={false}
            calendarClassName="calendar"
            calendarContainer={Container}
            disabledKeyboardNavigation
            calendarStartDay={1}
            useWeekdaysShort
            minDate={minDate}
            maxDate={maxDate}
            excludeDates={excludeDates}
            className="datepickerInput"
            inline={inline}
            selectsRange={selectsRange}
            shouldCloseOnSelect={!selectsRange}
            // eslint-disable-next-line react/jsx-props-no-spreading
            {...props}
          >
            {props.captions && renderDatepickerCaptions(props.captions)}
            {(props.primaryAction || props.secondaryAction) && renderActions()}
          </DatePicker>
        </StyledDatepicker>
      </StyledWrapper>
    </TypographyProvider>
  );
};
