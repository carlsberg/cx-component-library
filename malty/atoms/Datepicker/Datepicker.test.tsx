import { jsonRenderer } from '@carlsberggroup/malty.utils.test';
import React, { useState } from 'react';
import { Datepicker } from './Datepicker';

const label = 'Select date';

describe('datepicker', () => {
  it('matches snapshot', () => {
    const [startDate, setStartDate] = useState(new Date());
    const view = jsonRenderer(
      <Datepicker label={label} onChange={(date: Date) => setStartDate(date)} startDate={startDate} />
    );
    expect(view).toMatchSnapshot();
  });
});
