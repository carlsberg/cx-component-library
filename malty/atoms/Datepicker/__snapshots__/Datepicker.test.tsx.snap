// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datepicker matches snapshot 1`] = `
<span
  className="sc-bdvvtL iuuuBw"
>
  <div
    className="sc-dkPtRN hsXNXs"
  >
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <div
          className="sc-hKwDye jFAzKv"
        >
          <label
            htmlFor="datepicker-input"
            onClick={[Function]}
          >
            Select date
          </label>
          <button
            className="datepickerInput"
            name="datepicker-input"
            onClick={[Function]}
            type="button"
          >
            03/28/2022
            <svg
              className="sc-gsDKAQ kyUyaQ"
              color="digital-black"
              data-testid="svg-component"
              size="24px"
              version="1.1"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <g
                fillRule="evenodd"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17 1a1 1 0 01.993.883L18 2v1h2a3 3 0 012.995 2.824L23 6v13a3 3 0 01-3 3H4a3 3 0 01-3-3V6a3 3 0 013-3h2V2a1 1 0 011.993-.117L8 2v1h8V2a1 1 0 011-1zm4 9H3v9a1 1 0 00.883.993L4 20h16a1 1 0 001-1zM9 16a1 1 0 010 2H7a1 1 0 010-2zm8 0a1 1 0 010 2h-2a1 1 0 010-2zm-8-4a1 1 0 010 2H7a1 1 0 010-2zm8 0a1 1 0 010 2h-2a1 1 0 010-2zM6 5H4a1 1 0 00-1 1v2h18V6a1 1 0 00-.883-.993L20 5h-2v1a1 1 0 01-1.993.117L16 6V5H8v1a1 1 0 01-1.993.117L6 6z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</span>
`;
